### PR TITLE
CI fixups: python2 in azure, llvm 16 cmake dependency

### PR DESCRIPTION
**Pull request by eli-schwartz**
See:
- llvm
  #11642
- python2
  https://github.com/mesonbuild/meson/pull/11733#issuecomment-1572909430
  https://github.com/mesonbuild/meson/pull/11049#issuecomment-1577668311

### CODE REVIEW

**Review by tristan957 — APPROVED** [review]
Looks a-ok to me. Way to work on this one since I am sure it wasn't easy.

File: test cases/frameworks/15 llvm/meson.build [hunk 1]
```diff
@@ -3,23 +3,30 @@ project('llvmtest', ['c', 'cpp'], default_options : ['c_std=c99'])
...
-  d = dependency('llvm', version : static ? '>0.1' : '>=7.0', required : false, static : static)
-  if not d.found()
-    error('MESON_SKIP_TEST llvm not found or llvm version is too low')
+d = dependency('llvm', required : false, static : static)
+if not d.found()
+  error('MESON_SKIP_TEST llvm not found.')
+endif
+
+modules_to_find = [
+  'bitwriter', 'asmprinter', 'executionegine', 'mcjit', 'target',
```
Comment (eli-schwartz) [inline test cases/frameworks/15 llvm/meson.build:12]: Tests fail because of this typo. Should be executionengine. Will fix later today.

**Comment by jpakkane** [comment]
A related question is how much longer should be keep Python 2 support going? I would be fine with deleting it immediately but maybe there are still people using it?

**Comment by eli-schwartz** [comment]
I see no reason to delete the support. A better question is when to stop testing it in CI.

It takes no ongoing maintenance work to support it, other than to keep the introspection script capable of executing in a python2 interpreter. Which is *extremely* easy as there's no useful python3 specific features to want to use there (type annotations comes closest).

As far as testing in CI goes, a major benefit of this is that we actually do test the support for "building for two versions of python at the same time".

**Comment by dcbaker** [comment]
> Which is extremely easy as there's no useful python3 specific features to want to use there (type annotations comes closest).

And we can always use the oldschool comment style for that, which is clunky but not too bad for small bits of code. Mypy also has a python2 mode, so we could even teach ./run_mypy to test the introspection scripts in for python2 support

**Pull request merged**